packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -356,6 +356,7 @@ class Doctor {
     bool showPii = true,
     List<ValidatorTask>? startedValidatorTasks,
     bool sendEvent = true,
+    FlutterVersion? version,
```

Comment:
This new parameter is unused. What was the intent?

Reply:
Looking at the change to doctor.dart, it looks like this was added because `fetchTagsAndUpdate` no longer exist. However, this new `FlutterVersion` object is not accessible to anything downstream.

The result is that downstream users of `globals.flutterVersion` may get a stale `FlutterVersion` object. Unless we are okay with `flutter doctor` getting a stale version value, we either need to

* pass this all the way down (i.e. try to eliminate references to the global outside of context_runner.dart) or
* maintain the mutable nature of `FlutterVersion`.

Reply:
Good catch. I actually got stuck figuring out how to pass this down to the validator and then forgot. It's gonna be ugly but I have to find a way to hack this into the one validator that needs it.

Reply:
nvm, found an easy way to do this...

Reply:
making the change was easy, the test not so much :)

https://github.com/flutter/flutter/pull/124558/commits/6b51331dd96838164b434ebb861a72b1ea062644